ledger/common/convert/convert_test.go

```diff
@@ -69,3 +69,45 @@ func TestRegisterIDToLedgerKey(t *testing.T) {
 	key := convert.RegisterIDToLedgerKey(registerID)
 	require.Equal(t, expectedKey, key)
 }
+
+func TestPayloadToRegister(t *testing.T) {
+	t.Run("can convert", func(t *testing.T) {
+		value := []byte("value")
+		p := ledger.NewPayload(
+			ledger.NewKey(
+				[]ledger.KeyPart{
+					ledger.NewKeyPart(convert.KeyPartOwner, []byte("owner")),
+					ledger.NewKeyPart(convert.KeyPartKey, []byte("key")),
+				},
+			),
+			value,
+		)
+		regID, regValue, err := convert.PayloadToRegister(p)
+		require.NoError(t, err)
+		require.Equal(t, flow.NewRegisterID("owner", "key"), regID)
+		require.Equal(t, value, regValue)
+	})
+
+	t.Run("global key", func(t *testing.T) {
+		value := []byte("1")
+		p := ledger.NewPayload(
+			ledger.NewKey(
+				[]ledger.KeyPart{
+					ledger.NewKeyPart(convert.KeyPartOwner, []byte("")),
+					ledger.NewKeyPart(convert.KeyPartKey, []byte("uuid")),
+				},
+			),
+			value,
+		)
+		regID, regValue, err := convert.PayloadToRegister(p)
+		require.NoError(t, err)
+		require.Equal(t, flow.NewRegisterID("", "uuid"), regID)
```

Comment:
maybe add this to make sure it converted properly
```suggestion
		require.Equal(t, flow.NewRegisterID("", "uuid"), regID)
		require.True(t, regID.IsInternalState())
```